src/static/hyperhtml/documentation/index.html

```diff
@@ -630,10 +630,46 @@ <h3 id="essentials-6-2">The data Attribute</h3>
               <p>
                 The following code will log in console exactly the <code>user</code> reference.
                 No JSON is involved whatsoever in the process.
-              </p.>
+              </p>
               <pre><code class="javascript">
               hyper(ref)`&lt;p onconnected=${function () { console.log(this.data); }} data=${user} /p&gt;`;
               </code></pre>
+              <p>
+                If you however find yourself restricted by <code>data</code>, big news for you: hyperHTML doesn't lock you in and it lets you free to implement it as follows for instance.
+              </p>
+              <pre><code class="javascript">
+              customElements.define(
+                'h-welcome',
+                class HyperWelcome extends HTMLElement {
+                  constructor(...args) {
+                    super(...args);
+                    this.html = hyperHTML.bind(self);
```

Comment:
when you invoke `super` in **any** ECMAScrfipt class, it could return a different instance that what you'd expect.

Accordingly, you want to use such instance and return it to the outer world.

```js
const self = super(...args);
self.html = hyperHTML.bind(self);
return self;
```

This is a universal pattern that makes polyfill also work. As example, document-register-element polyfill passes the element that is being upgraded, and the super call would take care of enriching such element with all prototype descriptor the custom class has.

That's the only way to "upgrade" an element.

If you want to simplify this because you have transpilers that will do the dance for you (i.e. Babel would do the same once transpiled) it's also OK, so for c clarity sake you either use `self` everywhere and return it, or you use `this` but you have to bind it, while you are binding `self` in there that is the `window` object, not the current instance.

TL:DR please drop `self` as bind or use the example code I've written.

Reply:
Thanks for the explaination, I removed `self` in favor of `this` since it was a leftover of copy-paste.